### DISCUSSION TITLE
Using @JsonSubTypes() with Micronaut MongoDB and Native image

### DISCUSSION THREAD

**Discussion by DKolibar** [micronaut-projects/micronaut-data#2068] [Q&A] [answered]
Hello, is it possible to use @JsonSubTypes() with Micronaut MongoDB and Native image?

I have class:
```
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Getter
@Setter
@MappedEntity(CollectionNames.USER)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "userType")
@JsonSubTypes({
        @JsonSubTypes.Type(value = Student.class),
        @JsonSubTypes.Type(value = Supervisor.class)
})
@ReflectiveAccess
public abstract class User {

    @Id
    @GeneratedValue
    private ObjectId id;

   ....

}
```
and then Student and Supervisor classes which have some own specific fields. I defined controller method for saving all types of Users:
```
    @Post(uri = ManagerEndpointsPaths.NEW_USER, consumes = MediaType.APPLICATION_JSON, produces = MediaType.APPLICATION_JSON)
    public HttpResponse<User> saveNewUser(@Body final User newUser){
        return HttpResponse.created(userRepository.save(newUser));
    }
```
When I run app outside of native image everything works fine, but inside native image I get error `io.micronaut.serde.exceptions.SerdeException: Cannot find naming strategy: interface io.micronaut.data.document.serde.IdPropertyNamingStrategy`. There is very little information about known issues of native image combination with micronaut data mongodb on internet so I am asking here. Are JsonSubTypes even supported right now? If yes what am I missing

**Comment by graemerocher** [comment, marked as answer]
Seems like a bug, can probably be worked around with `@TypeHint(IdPropertyNamingStrategy.class)`, please file an issue and attach a reproducer

**Reply by DKolibar** [reply]
I added `@TypeHint(...)` annotation and it helped, I will create issue for this. Thanks for your answer

**Reply by kamalbhatia528** [reply]
```
@DKolibar, @graemerocher ,I have also tried its not working for me, Please help /

`@AllArgsConstructor
@NoArgsConstructor
@Serdeable
@Data
@Introspected
@MappedEntity
@TypeHint(IdPropertyNamingStrategy.class)
public class Bbox {
    @Id
    @GeneratedValue
    private ObjectId id;
}


`

`@Post
    @Status(HttpStatus.CREATED)
    Bbox createBbox(@QueryValue("mapProvider") @NotBlank @Valid MapProvider mapProvider, @Body @NotNull @NonNull @Valid Bbox bbox);`
```
